crates/resolver/src/config.rs

```diff
@@ -177,17 +177,15 @@ pub struct NameServerConfig {
     pub ip: IpAddr,
     /// Whether to trust `NXDOMAIN` responses from upstream nameservers.
     ///
-    /// When this is `true`, and an empty `NXDOMAIN` response or `NOERROR`
-    /// with an empty answers set is received, the
-    /// query will not be retried against other configured name servers if
+    /// When this is `true`, and an empty `NXDOMAIN` response or `NOERROR` with an empty answers
+    /// set is received, the query will not be retried against other configured name servers if
     /// the response has the Authoritative flag set.
```

Comment:
It looks like this last part of the docs is out of date, as `PoolState::try_send()` does not check the authoritative answer flag. (It would be strange if we got an AA=1 response, as I figure the resolver should almost always be talking to a recursive name server)

Reply:
i think this note about authoritativeness was never correct, actually - https://github.com/hickory-dns/hickory-dns/issues/1860 describes the SOA record in a negative answer (`NXDOMAIN` or `NOERROR` with no data) as indicating the response is authoritative, but it's just present for RFC 2308-style negative caching TTL communication. i'm also basing that on comments such as [this one](https://github.com/hickory-dns/hickory-dns/pull/1861/files#diff-90f333e308fae6ca92b573fb451e69b70155c72e3f8d1a7f0c836062d0e7b8d1R352) from 1861 but that PR is ultimately where this remark came from as well.